### PR TITLE
fix(napi): Don't run microtasks in napi_resolve_deferred

### PR DESCRIPTION
**Pull request by nathanwhit**
Fixes an incredibly obscure bug that causes parcel's file watcher to not get any file update notifications on macOS.

The issue was that the native addon was calling `napi_resolve_deferred`, but when we resolved the promise, v8 was running microtasks automatically. That executed JS, which called back into the native addon and broke the addon's assumption that the call wouldn't be reentrant.

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
Would be lovely to have a test, but I understand it's really hard. Nice find

**Review by littledivy — APPROVED** [review]
LGTM

File: cli/napi/js_native_api.rs [hunk 1]
```diff
@@ -3307,6 +3307,11 @@ fn napi_resolve_deferred(
   check_arg!(env, result);
   check_arg!(env, deferred);
 
+  // Make sure microtasks don't run and call back into JS
+  env
+    .scope()
+    .set_microtasks_policy(v8::MicrotasksPolicy::Explicit);
+
```
[the remark this answers is not in the document]
Comment (bartlomieju) [reply cli/napi/js_native_api.rs:3314]: Oh, that's a good point. We do need to restore it, otherwise other things will break.

**Review by bartlomieju — CHANGES_REQUESTED** [review]
https://github.com/denoland/deno/pull/25246/files#r1734711266

File: cli/napi/js_native_api.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nathanwhit) [reply cli/napi/js_native_api.rs:3314]: Oh good catch, fixed

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**